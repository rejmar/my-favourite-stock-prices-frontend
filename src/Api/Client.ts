/* tslint:disable */
/* eslint-disable */
//----------------------
// <auto-generated>
//     Generated using the NSwag toolchain v13.8.2.0 (NJsonSchema v10.2.1.0 (Newtonsoft.Json v11.0.0.0)) (http://NSwag.org)
// </auto-generated>
//----------------------
// ReSharper disable InconsistentNaming

export interface IClient {
  /**
   * Fetch all favourite stocks
   * @return OK
   */
  getAllFavStocks(): Promise<FavouriteStock[]>;
  /**
   * Add stock to favourites
   * @return OK
   */
  addNewFavStock(body: Stock): Promise<FavouriteStock>;
  /**
   * Delete stock from favourites
   * @param id Stock id
   * @return OK
   */
  deleteFavStock(id: number): Promise<FavouriteStock>;
  /**
   * Fetch all available stocks
   * @return OK
   */
  getAllStocks(): Promise<Stock[]>;
  /**
   * Get stock by name
   * @param name Stock's name
   * @return OK
   */
  getStockByName(name: string): Promise<Stock>;
}

export class Client implements IClient {
  private http: {
    fetch(url: RequestInfo, init?: RequestInit): Promise<Response>;
  };
  private baseUrl: string;
  protected jsonParseReviver:
    | ((key: string, value: any) => any)
    | undefined = undefined;

  constructor(
    baseUrl?: string,
    http?: { fetch(url: RequestInfo, init?: RequestInit): Promise<Response> }
  ) {
    this.http = http ? http : <any>window;
    this.baseUrl =
      baseUrl !== undefined && baseUrl !== null
        ? baseUrl
        : "http://localhost:8080";
  }

  /**
   * Fetch all favourite stocks
   * @return OK
   */
  getAllFavStocks(): Promise<FavouriteStock[]> {
    let url_ = this.baseUrl + "/v1/fav-stocks/";
    url_ = url_.replace(/[?&]$/, "");

    let options_ = <RequestInit>{
      method: "GET",
      headers: {
        Accept: "application/json",
      },
    };

    return this.http.fetch(url_, options_).then((_response: Response) => {
      return this.processGetAllFavStocks(_response);
    });
  }

  protected processGetAllFavStocks(
    response: Response
  ): Promise<FavouriteStock[]> {
    const status = response.status;
    let _headers: any = {};
    if (response.headers && response.headers.forEach) {
      response.headers.forEach((v: any, k: any) => (_headers[k] = v));
    }
    if (status === 404) {
      return response.text().then((_responseText) => {
        let result404: any = null;
        let resultData404 =
          _responseText === ""
            ? null
            : JSON.parse(_responseText, this.jsonParseReviver);
        result404 = resultData404 !== undefined ? resultData404 : <any>null;
        return throwException(
          "Not Found",
          status,
          _responseText,
          _headers,
          result404
        );
      });
    } else if (status === 400) {
      return response.text().then((_responseText) => {
        let result400: any = null;
        let resultData400 =
          _responseText === ""
            ? null
            : JSON.parse(_responseText, this.jsonParseReviver);
        result400 = resultData400 !== undefined ? resultData400 : <any>null;
        return throwException(
          "Bad Request",
          status,
          _responseText,
          _headers,
          result400
        );
      });
    } else if (status === 200) {
      return response.text().then((_responseText) => {
        let result200: any = null;
        let resultData200 =
          _responseText === ""
            ? null
            : JSON.parse(_responseText, this.jsonParseReviver);
        if (Array.isArray(resultData200)) {
          result200 = [] as any;
          for (let item of resultData200)
            result200!.push(FavouriteStock.fromJS(item));
        }
        return result200;
      });
    } else if (status !== 200 && status !== 204) {
      return response.text().then((_responseText) => {
        return throwException(
          "An unexpected server error occurred.",
          status,
          _responseText,
          _headers
        );
      });
    }
    return Promise.resolve<FavouriteStock[]>(<any>null);
  }

  /**
   * Add stock to favourites
   * @return OK
   */
  addNewFavStock(body: Stock): Promise<FavouriteStock> {
    let url_ = this.baseUrl + "/v1/fav-stocks/";
    url_ = url_.replace(/[?&]$/, "");

    const content_ = JSON.stringify(body);

    let options_ = <RequestInit>{
      body: content_,
      method: "POST",
      headers: {
        "Content-Type": "application/json",
        Accept: "application/json",
      },
    };

    return this.http.fetch(url_, options_).then((_response: Response) => {
      return this.processAddNewFavStock(_response);
    });
  }

  protected processAddNewFavStock(response: Response): Promise<FavouriteStock> {
    const status = response.status;
    let _headers: any = {};
    if (response.headers && response.headers.forEach) {
      response.headers.forEach((v: any, k: any) => (_headers[k] = v));
    }
    if (status === 404) {
      return response.text().then((_responseText) => {
        let result404: any = null;
        let resultData404 =
          _responseText === ""
            ? null
            : JSON.parse(_responseText, this.jsonParseReviver);
        result404 = resultData404 !== undefined ? resultData404 : <any>null;
        return throwException(
          "Not Found",
          status,
          _responseText,
          _headers,
          result404
        );
      });
    } else if (status === 400) {
      return response.text().then((_responseText) => {
        let result400: any = null;
        let resultData400 =
          _responseText === ""
            ? null
            : JSON.parse(_responseText, this.jsonParseReviver);
        result400 = resultData400 !== undefined ? resultData400 : <any>null;
        return throwException(
          "Bad Request",
          status,
          _responseText,
          _headers,
          result400
        );
      });
    } else if (status === 200) {
      return response.text().then((_responseText) => {
        let result200: any = null;
        let resultData200 =
          _responseText === ""
            ? null
            : JSON.parse(_responseText, this.jsonParseReviver);
        result200 = FavouriteStock.fromJS(resultData200);
        return result200;
      });
    } else if (status !== 200 && status !== 204) {
      return response.text().then((_responseText) => {
        return throwException(
          "An unexpected server error occurred.",
          status,
          _responseText,
          _headers
        );
      });
    }
    return Promise.resolve<FavouriteStock>(<any>null);
  }

  /**
   * Delete stock from favourites
   * @param id Stock id
   * @return OK
   */
  deleteFavStock(id: number): Promise<FavouriteStock> {
    let url_ = this.baseUrl + "/v1/fav-stocks/{id}";
    if (id === undefined || id === null)
      throw new Error("The parameter 'id' must be defined.");
    url_ = url_.replace("{id}", encodeURIComponent("" + id));
    url_ = url_.replace(/[?&]$/, "");

    let options_ = <RequestInit>{
      method: "DELETE",
      headers: {
        Accept: "application/json",
      },
    };

    return this.http.fetch(url_, options_).then((_response: Response) => {
      return this.processDeleteFavStock(_response);
    });
  }

  protected processDeleteFavStock(response: Response): Promise<FavouriteStock> {
    const status = response.status;
    let _headers: any = {};
    if (response.headers && response.headers.forEach) {
      response.headers.forEach((v: any, k: any) => (_headers[k] = v));
    }
    if (status === 404) {
      return response.text().then((_responseText) => {
        let result404: any = null;
        let resultData404 =
          _responseText === ""
            ? null
            : JSON.parse(_responseText, this.jsonParseReviver);
        result404 = resultData404 !== undefined ? resultData404 : <any>null;
        return throwException(
          "Not Found",
          status,
          _responseText,
          _headers,
          result404
        );
      });
    } else if (status === 400) {
      return response.text().then((_responseText) => {
        let result400: any = null;
        let resultData400 =
          _responseText === ""
            ? null
            : JSON.parse(_responseText, this.jsonParseReviver);
        result400 = resultData400 !== undefined ? resultData400 : <any>null;
        return throwException(
          "Bad Request",
          status,
          _responseText,
          _headers,
          result400
        );
      });
    } else if (status === 200) {
      return response.text().then((_responseText) => {
        let result200: any = null;
        let resultData200 =
          _responseText === ""
            ? null
            : JSON.parse(_responseText, this.jsonParseReviver);
        result200 = FavouriteStock.fromJS(resultData200);
        return result200;
      });
    } else if (status !== 200 && status !== 204) {
      return response.text().then((_responseText) => {
        return throwException(
          "An unexpected server error occurred.",
          status,
          _responseText,
          _headers
        );
      });
    }
    return Promise.resolve<FavouriteStock>(<any>null);
  }

  /**
   * Fetch all available stocks
   * @return OK
   */
  getAllStocks(): Promise<Stock[]> {
    let url_ = this.baseUrl + "/v1/stocks/";
    url_ = url_.replace(/[?&]$/, "");

    let options_ = <RequestInit>{
      method: "GET",
      headers: {
        Accept: "application/json",
      },
    };

    return this.http.fetch(url_, options_).then((_response: Response) => {
      return this.processGetAllStocks(_response);
    });
  }

  protected processGetAllStocks(response: Response): Promise<Stock[]> {
    const status = response.status;
    let _headers: any = {};
    if (response.headers && response.headers.forEach) {
      response.headers.forEach((v: any, k: any) => (_headers[k] = v));
    }
    if (status === 404) {
      return response.text().then((_responseText) => {
        let result404: any = null;
        let resultData404 =
          _responseText === ""
            ? null
            : JSON.parse(_responseText, this.jsonParseReviver);
        result404 = resultData404 !== undefined ? resultData404 : <any>null;
        return throwException(
          "Not Found",
          status,
          _responseText,
          _headers,
          result404
        );
      });
    } else if (status === 400) {
      return response.text().then((_responseText) => {
        let result400: any = null;
        let resultData400 =
          _responseText === ""
            ? null
            : JSON.parse(_responseText, this.jsonParseReviver);
        result400 = resultData400 !== undefined ? resultData400 : <any>null;
        return throwException(
          "Bad Request",
          status,
          _responseText,
          _headers,
          result400
        );
      });
    } else if (status === 200) {
      return response.text().then((_responseText) => {
        let result200: any = null;
        let resultData200 =
          _responseText === ""
            ? null
            : JSON.parse(_responseText, this.jsonParseReviver);
        if (Array.isArray(resultData200)) {
          result200 = [] as any;
          for (let item of resultData200) result200!.push(Stock.fromJS(item));
        }
        return result200;
      });
    } else if (status !== 200 && status !== 204) {
      return response.text().then((_responseText) => {
        return throwException(
          "An unexpected server error occurred.",
          status,
          _responseText,
          _headers
        );
      });
    }
    return Promise.resolve<Stock[]>(<any>null);
  }

  /**
   * Get stock by name
   * @param name Stock's name
   * @return OK
   */
  getStockByName(name: string): Promise<Stock> {
    let url_ = this.baseUrl + "/v1/stocks/{name}";
    if (name === undefined || name === null)
      throw new Error("The parameter 'name' must be defined.");
    url_ = url_.replace("{name}", encodeURIComponent("" + name));
    url_ = url_.replace(/[?&]$/, "");

    let options_ = <RequestInit>{
      method: "GET",
      headers: {
        Accept: "application/json",
      },
    };

    return this.http.fetch(url_, options_).then((_response: Response) => {
      return this.processGetStockByName(_response);
    });
  }

  protected processGetStockByName(response: Response): Promise<Stock> {
    const status = response.status;
    let _headers: any = {};
    if (response.headers && response.headers.forEach) {
      response.headers.forEach((v: any, k: any) => (_headers[k] = v));
    }
    if (status === 404) {
      return response.text().then((_responseText) => {
        let result404: any = null;
        let resultData404 =
          _responseText === ""
            ? null
            : JSON.parse(_responseText, this.jsonParseReviver);
        result404 = resultData404 !== undefined ? resultData404 : <any>null;
        return throwException(
          "Not Found",
          status,
          _responseText,
          _headers,
          result404
        );
      });
    } else if (status === 400) {
      return response.text().then((_responseText) => {
        let result400: any = null;
        let resultData400 =
          _responseText === ""
            ? null
            : JSON.parse(_responseText, this.jsonParseReviver);
        result400 = resultData400 !== undefined ? resultData400 : <any>null;
        return throwException(
          "Bad Request",
          status,
          _responseText,
          _headers,
          result400
        );
      });
    } else if (status === 200) {
      return response.text().then((_responseText) => {
        let result200: any = null;
        let resultData200 =
          _responseText === ""
            ? null
            : JSON.parse(_responseText, this.jsonParseReviver);
        result200 = Stock.fromJS(resultData200);
        return result200;
      });
    } else if (status !== 200 && status !== 204) {
      return response.text().then((_responseText) => {
        return throwException(
          "An unexpected server error occurred.",
          status,
          _responseText,
          _headers
        );
      });
    }
    return Promise.resolve<Stock>(<any>null);
  }
}

/** Stock to add to favourites */
export class Stock implements IStock {
  id!: number;
  name!: string;
  price!: number;

  constructor(data?: IStock) {
    if (data) {
      for (var property in data) {
        if (data.hasOwnProperty(property))
          (<any>this)[property] = (<any>data)[property];
      }
    }
  }

  init(_data?: any) {
    if (_data) {
      this.id = _data["id"];
      this.name = _data["name"];
      this.price = _data["price"];
    }
  }

  static fromJS(data: any): Stock {
    data = typeof data === "object" ? data : {};
    let result = new Stock();
    result.init(data);
    return result;
  }

  toJSON(data?: any) {
    data = typeof data === "object" ? data : {};
    data["id"] = this.id;
    data["name"] = this.name;
    data["price"] = this.price;
    return data;
  }
}

/** Stock to add to favourites */
export interface IStock {
  id: number;
  name: string;
  price: number;
}

export class FavouriteStock implements IFavouriteStock {
  id!: number;
  name!: string;
  price!: number;
  timestamp!: string;

  constructor(data?: IFavouriteStock) {
    if (data) {
      for (var property in data) {
        if (data.hasOwnProperty(property))
          (<any>this)[property] = (<any>data)[property];
      }
    }
  }

  init(_data?: any) {
    if (_data) {
      this.id = _data["id"];
      this.name = _data["name"];
      this.price = _data["price"];
      this.timestamp = _data["timestamp"];
    }
  }

  static fromJS(data: any): FavouriteStock {
    data = typeof data === "object" ? data : {};
    let result = new FavouriteStock();
    result.init(data);
    return result;
  }

  toJSON(data?: any) {
    data = typeof data === "object" ? data : {};
    data["id"] = this.id;
    data["name"] = this.name;
    data["price"] = this.price;
    data["timestamp"] = this.timestamp;
    return data;
  }
}

export interface IFavouriteStock {
  id: number;
  name: string;
  price: number;
  timestamp: string;
}

export class ApiException extends Error {
  message: string;
  status: number;
  response: string;
  headers: { [key: string]: any };
  result: any;

  constructor(
    message: string,
    status: number,
    response: string,
    headers: { [key: string]: any },
    result: any
  ) {
    super();

    this.message = message;
    this.status = status;
    this.response = response;
    this.headers = headers;
    this.result = result;
  }

  protected isApiException = true;

  static isApiException(obj: any): obj is ApiException {
    return obj.isApiException === true;
  }
}

function throwException(
  message: string,
  status: number,
  response: string,
  headers: { [key: string]: any },
  result?: any
): any {
  if (result !== null && result !== undefined) throw result;
  else throw new ApiException(message, status, response, headers, null);
}
