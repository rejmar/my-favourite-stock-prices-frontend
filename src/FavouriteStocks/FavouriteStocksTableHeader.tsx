import classNames from "classnames";
import React, { FunctionComponent } from "react";

const FavouriteStocksTableHeader: FunctionComponent = () => {
  return (
    <thead
      data-testid="stocks-table__header"
      className={classNames("thead-dark")}
    >
      <tr className={classNames("d-flex")}>
        <th scope="col" className={classNames("col-6")}>
          Stock Name
        </th>
        <th scope="col" className={classNames("col-2")}>
          Timestamp
        </th>
        <th scope="col" className={classNames("col-2")}>
          Price
        </th>
        <th scope="col" className={classNames("col-2")}>
          Action
        </th>
      </tr>
    </thead>
  );
};

export default FavouriteStocksTableHeader;
