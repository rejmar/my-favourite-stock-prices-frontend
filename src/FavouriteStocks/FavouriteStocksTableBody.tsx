import React from "react";
import { FavouriteStock } from "../Api/Client";
import FavouriteStockTableRow from "./FavouriteStocksTableRow";

type StocksTableBodyProps = {
  data: FavouriteStock[];
  deleteBookmark: (stockId: number) => void;
};
const FavouriteStocksTableBody: React.FunctionComponent<StocksTableBodyProps> = ({
  data,
  deleteBookmark,
}: StocksTableBodyProps) => {
  return (
    <tbody data-testid={"favourite-stocks-table__body"}>
      {data.map((stock: FavouriteStock) => (
        <FavouriteStockTableRow
          key={`${stock.price}`}
          stock={stock}
          deleteFromBookmarks={deleteBookmark}
        />
      ))}
    </tbody>
  );
};

export default FavouriteStocksTableBody;
