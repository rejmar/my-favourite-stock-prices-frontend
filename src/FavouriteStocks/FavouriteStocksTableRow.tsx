import classNames from "classnames";
import React from "react";
import Button from "react-bootstrap/Button";
import { FavouriteStock } from "../Api/Client";
import { getStockNameFromStockCode, roundPrice } from "../utils";

type StockTableRowProps = {
  stock: FavouriteStock;
  deleteFromBookmarks: (stockId: number) => void;
};

const FavouriteStockTableRow: React.FunctionComponent<StockTableRowProps> = ({
  stock: favouriteStock,
  deleteFromBookmarks,
}: StockTableRowProps) => {
  return (
    <tr
      data-testid={`favourite-stock-table__row__${favouriteStock.name}`}
      className={classNames("d-flex")}
    >
      <td
        data-testid={`favourite-stock-table__row__${favouriteStock.name}__name`}
        className={classNames("col-6")}
      >
        {getStockNameFromStockCode(favouriteStock.name) ?? ""}
      </td>
      <td
        data-testid={`favourite-stock-table__row__${favouriteStock.name}__timestamp`}
        className={classNames("col-2")}
      >
        {favouriteStock.timestamp}
      </td>
      <td
        data-testid={`favourite-stock-table__row__${favouriteStock.name}__price`}
        className={classNames("col-2")}
      >
        {roundPrice(favouriteStock.price ?? 0) ?? ""}
      </td>
      <td
        data-testid={`favourite-stock-table__row__${favouriteStock.name}__action`}
        className={classNames("d-flex", "col-2", "justify-content-center")}
      >
        <Button
          className={classNames("p-1", "border-secondary", "bg-secondary")}
          onClick={() => deleteFromBookmarks(favouriteStock.id)}
        >
          Remove Bookmark!
        </Button>
      </td>
    </tr>
  );
};

export default FavouriteStockTableRow;
