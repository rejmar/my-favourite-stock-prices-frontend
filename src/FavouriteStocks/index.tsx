import classNames from "classnames";
import React from "react";
import Table from "react-bootstrap/Table";
import FavouriteStocksTableHeader from "./FavouriteStocksTableHeader";
import FavouriteStocksTableBody from "./FavouriteStocksTableBody";
import { FavouriteStock } from "../Api/Client";

type FavouriteStocksProps = {
  data: FavouriteStock[];
  deleteBookmark: (stockId: number) => void;
};
const FavouriteStocks: React.FunctionComponent<FavouriteStocksProps> = ({
  data,
  deleteBookmark,
}: FavouriteStocksProps) => {
  return (
    <div
      data-testid="favourite-stocks"
      className={classNames(
        "d-flex",
        "flex-column",
        "w-100",
        "border",
        "border-light",
        "rounded",
        "bg-light",
        "p-2"
      )}
    >
      <h2 data-testid="favourite-stocks__label">Bookmarks</h2>
      <div
        data-testid="favourite-stocks-table"
        className={classNames("d-flex", "w-100")}
      >
        <Table striped bordered hover>
          <FavouriteStocksTableHeader />
          <FavouriteStocksTableBody
            data={data}
            deleteBookmark={deleteBookmark}
          />
        </Table>
      </div>
    </div>
  );
};

export default FavouriteStocks;
