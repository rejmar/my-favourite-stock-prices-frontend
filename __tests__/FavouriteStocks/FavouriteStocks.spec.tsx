import React from "react";
import renderer from "react-test-renderer";
import { FavouriteStock } from "../../src/Api/Client";
import FavouriteStocks from "../../src/FavouriteStocks";

describe("FavouriteStocks tests", () => {
  it("matches snapshot", () => {
    const stocksMock: FavouriteStock[] = [
      new FavouriteStock({
        id: 1,
        name: "APPLE",
        timestamp: "01/01/2000 01:01",
        price: 100,
      }),
      new FavouriteStock({
        id: 2,
        name: "MICROSOFT",
        timestamp: "01/01/2000 01:01",
        price: 200,
      }),
    ];
    const deleteActionMock: jest.Mock = jest.fn();

    const tree = renderer
      .create(
        <FavouriteStocks data={stocksMock} deleteBookmark={deleteActionMock} />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
