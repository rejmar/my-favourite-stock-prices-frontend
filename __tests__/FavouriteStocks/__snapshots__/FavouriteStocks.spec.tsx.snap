// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavouriteStocks tests matches snapshot 1`] = `
<div
  className="d-flex flex-column w-100 border border-light rounded bg-light p-2"
  data-testid="favourite-stocks"
>
  <h2
    data-testid="favourite-stocks__label"
  >
    Bookmarks
  </h2>
  <div
    className="d-flex w-100"
    data-testid="favourite-stocks-table"
  >
    <table
      className="table table-striped table-bordered table-hover"
    >
      <thead
        className="thead-dark"
        data-testid="stocks-table__header"
      >
        <tr
          className="d-flex"
        >
          <th
            className="col-6"
            scope="col"
          >
            Stock Name
          </th>
          <th
            className="col-2"
            scope="col"
          >
            Timestamp
          </th>
          <th
            className="col-2"
            scope="col"
          >
            Price
          </th>
          <th
            className="col-2"
            scope="col"
          >
            Action
          </th>
        </tr>
      </thead>
      <tbody
        data-testid="favourite-stocks-table__body"
      >
        <tr
          className="d-flex"
          data-testid="favourite-stock-table__row__APPLE"
        >
          <td
            className="col-6"
            data-testid="favourite-stock-table__row__APPLE__name"
          >
            Apple
          </td>
          <td
            className="col-2"
            data-testid="favourite-stock-table__row__APPLE__timestamp"
          >
            01/01/2000 01:01
          </td>
          <td
            className="col-2"
            data-testid="favourite-stock-table__row__APPLE__price"
          >
            100.00
          </td>
          <td
            className="d-flex col-2 justify-content-center"
            data-testid="favourite-stock-table__row__APPLE__action"
          >
            <button
              className="p-1 border-secondary bg-secondary btn btn-primary"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Remove Bookmark!
            </button>
          </td>
        </tr>
        <tr
          className="d-flex"
          data-testid="favourite-stock-table__row__MICROSOFT"
        >
          <td
            className="col-6"
            data-testid="favourite-stock-table__row__MICROSOFT__name"
          >
            Microsoft
          </td>
          <td
            className="col-2"
            data-testid="favourite-stock-table__row__MICROSOFT__timestamp"
          >
            01/01/2000 01:01
          </td>
          <td
            className="col-2"
            data-testid="favourite-stock-table__row__MICROSOFT__price"
          >
            200.00
          </td>
          <td
            className="d-flex col-2 justify-content-center"
            data-testid="favourite-stock-table__row__MICROSOFT__action"
          >
            <button
              className="p-1 border-secondary bg-secondary btn btn-primary"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Remove Bookmark!
            </button>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
